### PR TITLE
Propagate handler name to wrapper

### LINKED ISSUE #146: dd-trace integration: missing resource name
We're using `express` and `express-promise-router` along with `dd-trace` to trace our Express requests with Datadog.

`express-promise-router` is creating [a wrapper function](https://github.com/express-promise-router/express-promise-router/blob/trunk/lib/express-promise-router.js#L51-L60) around the handler that is anonymous, meaning that it doesn't have a function name and loses the function name data from the original handler.

This results in a resource name of `<anonymous>` in `dd-trace` because there isn't a function name ([how dd-trace uses the function name](https://github.com/DataDog/dd-trace-js/blob/8eaaebc70e619fae91c97c23d925d019896508c7/packages/dd-trace/src/plugins/util/web.js#L106)), making it harder to debug problems.

Is it reasonable to modify `express-promise-router` to ensure that the wrapper functions have the same function name as the original handler function? Something like:
```
  var wrapperObj = {};
  if (handler.length === 4) {
    wrapperObj[handler.name] = function (err, req, res, next) {
      handleReturn([err, req, res, next]);
    };
  } else {
    wrapperObj[handler.name] = function (req, res, next) {
      handleReturn([req, res, next]);
    };
  }
  return wrapperObj[handler.name];
```

### PR DESCRIPTION
**Pull request by mormahr**
We do this to improve debuggability and tracing support.
Fixes #146.

Hey @mwinstanley, please check if this solves your problem.

### CODE REVIEW

File: test/tracing.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const Router = require("../lib/express-promise-router");
+const assert = require("chai").assert;
+
+describe("function name", function () {
+  it("uses the wrapped handler's name in a route handler", function () {
```
Comment (mwinstanley) [inline test/tracing.test.js:5]: Can you add a test case to make sure that everything works as expected if the function is actually anonymous? eg the name should be empty?

**Review by mwinstanley — APPROVED** [review]
Thank you so much!

File: test/tracing.test.js (hunk 1 above)
Comment (mormahr) [reply test/tracing.test.js:5]: Yeah that seems like a good idea!

**Comment by mormahr** [comment]
Hey @mwinstanley could you check if the added tests resolve your comments?
I think otherwise I'd just merge this as-is.

**Comment by mwinstanley** [comment]
Hey @mormahr I think this is fine, but I meant a test like this (haven't actually run this, don't take this as truth - but the idea is testing what happens if the function is a lambda):
```
  it("empty name for anonymous middleware handler", function () {
    const router = Router();
    router.use((req, res, next) => {});

    assert.equal(router.stack[0].handle.name, "");
    assert.equal(router.stack[0].name, "");
  });
```

**Comment by mormahr** [comment]
Like i said the way this currently plays out is this:

```js
assert.equal(router.stack[0].handle.name, "");
assert.equal(router.stack[0].name, "<anonymous>");
```

I might just a test so this doesn't regress to something else, but I think this is probably the correct behaviour?

**Pull request merged**